src/shared/chatbot/chatbot.service.ts

```diff
@@ -14,24 +15,46 @@ export class ChatbotService {
   private _chatbots: Map<string, Chatbot> = new Map();
   constructor(
     private readonly _logger: Logger,
-    private readonly _chainBuilderService: ChainBuilderService,
+    private readonly _langchainChainService: LangchainChainService,
+    private readonly _langchainService: LangchainService,
   ) {}
 
-  public async createChatbot(chatbotId: string | number, modelName?: string) {
-    const chain = await this._chainBuilderService.createChain(modelName);
+  public async createChatbot(chatbotId: string, modelName?: string) {
+    this._logger.log(
+      `Creating chatbot with id: ${chatbotId} and model: ${modelName}`,
+    );
+    const chain =
+      await this._langchainService.createMongoMemoryChatChain(modelName);
 
+    // @ts-ignore
     const chatbot = { chain } as Chatbot;
-    this._chatbots.set(String(chatbotId), chatbot);
+    this._chatbots.set(chatbotId, chatbot);
```

Comment:
**Suggestion:** Consider removing the `// @ts-ignore` comments and properly type the `chatbot` object. This will improve type safety and reduce the risk of runtime errors. [best practice]
```suggestion
    const chatbot: Chatbot = { chain };
    this._chatbots.set(chatbotId, chatbot);
```

Reply:
explain